src/manager/TickManager.ts

```diff
@@ -97,8 +105,14 @@ export class TickManager {
     feeGrowthGlobal0X128: JSBI,
     feeGrowthGlobal1X128: JSBI
   ): { feeGrowthInside0X128: JSBI; feeGrowthInside1X128: JSBI } {
-    const lower = this.get(tickLower);
-    const upper = this.get(tickUpper);
+    assert(
+      this.sortedTicks.has(tickLower) &&
+        this.sortedTicks.has(tickUpper) &&
+        this.sortedTicks.has(tickCurrent),
```

Comment:
The logic here is a good catch to me, but this line, `tickCurrent` is global tickIndex which can be any tick according to swap condition, not specifically a tick as position endpoint, so it could not existed in the `sortedTicks`.